sdk/core/Azure.Core.Experimental/src/Variant/Variant.cs

```diff
@@ -1102,17 +1102,19 @@ public readonly bool TryGetValue<T>(out T value)
             }
             else if (typeof(T) == typeof(DateTime) && _object == TypeFlags.DateTime)
             {
-                value = Unsafe.As<DateTime, T>(ref Unsafe.AsRef(_union.DateTime));
+                value = Unsafe.As<DateTime, T>(ref Unsafe.AsRef(in _union.DateTime));
                 success = true;
             }
             else if (typeof(T) == typeof(DateTimeOffset) && _object == TypeFlags.DateTimeOffset)
             {
-                value = Unsafe.As<DateTimeOffset, T>(ref Unsafe.AsRef(new DateTimeOffset(_union.Ticks, TimeSpan.Zero)));
+                var dateTimeOffset = new DateTimeOffset(_union.Ticks, TimeSpan.Zero);
```

Comment:
See: [CS9193: Argument should be a variable because it is passed to a ref readonly parameter](https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/compiler-messages/ref-modifiers-errors?f1url=%3FappId%3Droslyn%26k%3Dk(CS9191)#reference-variables-require-a-referent)